libsolidity/ast/AST.h

```diff
@@ -978,18 +978,19 @@ class ModifierDefinition: public CallableDeclaration, public StructurallyDocumen
 		ASTPointer<ParameterList> const& _parameters,
 		bool _isVirtual,
 		ASTPointer<OverrideSpecifier> const& _overrides,
-		ASTPointer<Block> _body
+		ASTPointer<Block> const& _body
```

Comment:
Please keep passing by value here and std::move from it.

Reply:
You make a good point. But clang-tidy `modernize-pass-by-value` actually didn't make this suggestion for `FunctionDefinition.` I made this comment in the PR (but too late, https://github.com/ethereum/solidity/pull/8571)

A potential problem with `std::move` is that if `m_body(std::move(_body))` happens first, then we will likely get into trouble with `ImplementationOptional(_body != nullptr)` The rules for initialization lists does say that `ImplementationOptional(_body != nullptr)` will happen first; perhaps clang-tidy doesn't check the order and wants to play it safe.

If you don't see any issues, I can do the `std::move.` Additionaly also for `FunctionDefinition`

Reply:
@chriseth  I need a decision here.

Reply:
Right, I didn't see that! Better to keep it as you wrote it.